End-to-End-Tests.md

```diff
@@ -89,24 +85,16 @@ Finally, flakes mean that developers rerun failing tests more readily. We even i
 ### Preventing flakes
 
 Conceptually, preventing flakes is easy. We can use `waitFor` statements to make the tests deterministic despite testing a non-deterministic system. For example, suppose we have a function `waitForModal()` that waits for a modal to appear. Then we could write our test like this:
-
-```text
-               <---A--->
-
-                        +-------+
-                        | Modal |
-+----------+   +---//---+ opens +---------------------------------+
-| Click to |   |        +-------+                                 |
-| open     +---+                                                  +---->
-| modal    |   |        +----------------+    +-------------+     |
-+----------+   +---//---+ waitForModal() +-//-+ Click to    +-----+
-                        +----------------+    | close modal |
-                                              +-------------+
-
-               <---B---><-------C-------->
-
-
---------------------- time -------------------------------------------->
+```mermaid
```

Comment:
@U8NWXD can you please review if this image conveys the information accurately?

Reply:
It doesn't. The A, B, and C lines are meant to refer to parts of the flowchart, but in the Mermaid diagram they don't line up with the right parts of the flowchart anymore

Reply:
@gp201 - Mermaid is great for simple and straightforward diagrams. But it's limited when adding some "rich" features to the diagrams.

Reply:
@gp201 There are only two diagrams. The first one is this diagram that U8NWXD said does not convey the information accurately and the second one is the example I have given above.

Reply:
@deadex-ng I think @gp201 is asking that those two diagrams be listed in #81 so that we don't forget about them

Reply:
@gp201 and @U8NWXD I have done that.

> @deadex-ng I think @gp201 is asking that those two diagrams be listed in #81 so that we don't forget about them

Reply:
@deadex-ng Nice! Now can you undo the change for the figures in this PR? We will migrate the images to mermaid when we find a better solution. We don't want to confuse contributors so we will be using the previous image itself.

Reply:
@gp201 I have undone the changes